rpcs3/rpcs3qt/gui_application.cpp

```diff
@@ -402,12 +402,30 @@ void gui_application::InitializeCallbacks()
 		return nullptr;
 	};
 
+	// Second argument is to pass a type
+	static const auto make_qt_object = [] <typename T> (gui_application* _this, T*) -> std::shared_ptr<T>
+	{
+		std::shared_ptr<T> result;
+
+		if (!_this->m_show_gui)
+		{
+			return result;
+		}
+
+		Emu.BlockingCallFromMainThread([&result]()
+		{
+			result = std::make_shared<T>();
+		});
+
+		return result;
+	};
+
 	callbacks.get_gs_frame    = [this]() -> std::unique_ptr<GSFrameBase> { return get_gs_frame(); };
-	callbacks.get_msg_dialog  = [this]() -> std::shared_ptr<MsgDialogBase> { return m_show_gui ? std::make_shared<msg_dialog_frame>() : nullptr; };
-	callbacks.get_osk_dialog  = [this]() -> std::shared_ptr<OskDialogBase> { return m_show_gui ? std::make_shared<osk_dialog_frame>() : nullptr; };
+	callbacks.get_msg_dialog  = [this]() -> std::shared_ptr<MsgDialogBase> { return make_qt_object(this, static_cast<msg_dialog_frame*>(nullptr)); };
+	callbacks.get_osk_dialog  = [this]() -> std::shared_ptr<OskDialogBase> { return make_qt_object(this, static_cast<osk_dialog_frame*>(nullptr)); };
 	callbacks.get_save_dialog = []() -> std::unique_ptr<SaveDialogBase> { return std::make_unique<save_data_dialog>(); };
-	callbacks.get_sendmessage_dialog = [this]() -> std::shared_ptr<SendMessageDialogBase> { return std::make_shared<sendmessage_dialog_frame>(); };
-	callbacks.get_recvmessage_dialog = [this]() -> std::shared_ptr<RecvMessageDialogBase> { return std::make_shared<recvmessage_dialog_frame>(); };
+	callbacks.get_sendmessage_dialog = [this]() -> std::shared_ptr<SendMessageDialogBase> { return make_qt_object(this, static_cast<sendmessage_dialog_frame*>(nullptr)); };
+	callbacks.get_recvmessage_dialog = [this]() -> std::shared_ptr<RecvMessageDialogBase> { return make_qt_object(this, static_cast<recvmessage_dialog_frame*>(nullptr)); };
```

Comment:
these callbacks are already used in a blocking call.
So remove it in either place